lib/ui/window.dart

```diff
@@ -286,31 +543,197 @@ class Locale {
     }
   }
 
+  /// Unicode language variant codes.
+  ///
+  /// This iterable provides variants normalized to lowercase and in sorted
+  /// order, as per the Unicode LDML specification.
+  Iterable<String> get variants => _variants ?? <String>[];
+
+  // The private _variants field must have variants in lowercase and already
+  // sorted: constructors must construct it as such.
+  final collection.LinkedHashSet<String> _variants;
+
+  // A map representing all Locale Identifier extensions.
+  //
+  // Keys in this ordered map must be sorted, and both keys and values must all
+  // be lowercase: constructors must construct it as such.
+  //
+  // This map simultaneously reprents T-extensions, U-extensions, other
+  // extensions and the private use extensions. Implementation detailsf:
+  //
+  // * The 't' entry represents the optional "tlang" identifier of the T
+  //   Extension. If the T Extension is present but has no tlang value, the 't'
+  //   map entry's value must be an empty string.
+  // * The 'u' entry represents the optional attributes of the U Extension. They
+  //   must be sorted in alphabetical order, separated by hyphens, and be
+  //   lowercase. If the U Extension is present but has no attributes, the 'u'
+  //   map entry's value must be an empty string.
+  // * U-Extension keyword keys, matching _reKey, and T-Extension fields in the
+  //   map, whos field separator subtags match _reTkey, are directly entered
+  //   into this map. (These regular expressions don't match the same keys.)
+  // * Other singletons are entered directly into the map, with all
+  //   values/attributes associated with that singleton as the map entry's
+  //   value.
+  final collection.LinkedHashMap<String, String> _extensions;
+
+  /// Produces the Unicode BCP47 Locale Identifier for this locale.
+  ///
+  /// If the const constructor was used with bad parameters, the result might
```

Comment:
Changed it to "unnamed constructor", it sounded like that's the proper name of such constructors in Dart.

Reply:
I'd probably just say "if the object was created with bad values", since it could be created with bad values any number of ways, not just with the unnamed constructor.

Reply:
Would it not be a failure in API design to add more ways of creating bad-valued Locale instances? I would want this const-constructor to be the only dangerous constructor that breaks API promises. (I'd actually like to add asserts to the unnamed constructor, so we can be more secure in our API promises...)